pkg/spec/createconfig.go

```diff
@@ -518,11 +523,9 @@ func (c *CreateConfig) GetContainerCreateOptions(runtime *libpod.Runtime, pod *l
 	if c.CgroupParent != "" {
 		options = append(options, libpod.WithCgroupParent(c.CgroupParent))
 	}
-	// For a rootless container always cleanup the storage/network as they
-	// run in a different namespace thus not reusable when we restart.
-	if c.Detach || rootless.IsRootless() {
-		options = append(options, libpod.WithExitCommand(c.createExitCommand()))
-	}
+
+	// Always use a cleanup process to clean up Podman after termination
+	options = append(options, libpod.WithExitCommand(c.createExitCommand()))
```

Comment:
@mheon i have to ask ... worst case, what are the performance implications here?

Reply:
I don't think we've ever measured them.

This requires that we spin up a fresh Podman process, but we only do it after the container is already dead, and it's purely in the background - on a lightly-loaded system, the user notices no changes, Podman runs just as fast if not slightly faster.

On a heavily loaded system, we're introducing another fork/exec of Podman, a fairly heavyweight process, once per container runtime. This could be a perf concern on systems that see heavy container churn (starting/stopping very often).

Reply:
(Though, honestly, we're already using cleanup for all detached containers, and if you're running hundreds of high-churn containers, you're probably using detach...)